### ISSUE TITLE
Strange Issue with Variable

### ISSUE THREAD

**Issue by inidona** [jenkinsci/lockable-resources-plugin#130]
Hi,

we are using a scripted pipeline with following snippets
 lock(label: "MyLabel", quantity: 1, variable: 'MyVar') {
                   echo "Content of MyVar: ${MyVar}"
....

works fine if jenkins not waiting for lock. MyVar would be printed to console output.
But when jenkins have to wait some seconds we see the exception that MyVar is no property
groovy.lang.MissingPropertyException: No such property: MyVar  for class: WorkflowScript.

We updated Plugin and Jenkins to the current version. With Jenkins ver. 2.138.1 and Plugin 2.3 everything works fine.

did you have an idea ?
For me it looks like a timing issue.

thanks
andreas

**Comment by CheukLeung** [comment]
Hi,
I have a similar issue but very specifically to manually reserved resource.

```
lock(label: 'myLabel', quantity: 1, variable: "myLock" ) {
  println("myLcok: ${myLock}")
}
```
If I manually reserved all resources with `myLabel` and this lock is waiting, the variable would be empty (`No such property: myLock`) when I unreserved the resources manually.

Andreas, is your case similar or does it happen also when they are only reserving and unreserving using the pipeline syntax?

Regards,
Cheuk
